src/porepy/models/run_models.py

```diff
@@ -249,3 +233,22 @@ def time_step() -> None:
                 time_progressbar.update(n=model._time_step / initial_time_step)
 
     model.after_simulation()
+
+
+def _choose_solver(model, params: dict) -> Union[pp.LinearSolver, pp.NewtonSolver]:
+    """Choose between linear and non-linear solver.
+
+    Parameters:
+        model: Model class containing all information on parameters, variables,
+            discretization, geometry. Various methods such as those relating to solving
+            the system, see the appropriate solver for documentation.
+        params: Parameters related to the solution procedure.
+
+    """
```

Comment:
I think the documentation is fair for the time being. It should be updated as we gradually understand how the interface to the solver should be (as we have managed to do with the methods `prepare_simulation`, `before_nonlinear_iteration` etc. in the models), but at this stage, such abstractions are premature - I just felt like making a note of this somewhere.

Reply:
Sure! I have two minor suggestions, please commit and merge if you approve.